model_mommy/mommy.py

```diff
@@ -336,6 +336,17 @@ def _clean_attrs(self, attrs):
         def is_rel_field(x):
             return '__' in x
         self.fill_in_optional = attrs.pop('_fill_optional', False)
+        # error for non existing fields
+        if isinstance(self.fill_in_optional, (tuple, list, set)):
+            # parents and relations
+            wrong_fields = set(self.fill_in_optional) - set(
+                f.name for f in self.get_fields()
+            )
+            if wrong_fields:
+                raise AttributeError(
+                    '_fill_optional field(s) %s are not related to model %s'
+                    % (list(wrong_fields), self.model.__name__)
+                )
```

Comment:
I wonder if we can make this part a bit easier for contributors to understand.
I've tinkered something like this from top of my head:

```python
    def _clean_attrs(self, attrs):
        def is_rel_field(x):
            return '__' in x

        def field_exists(field):
            return field in [f.name for f in self.get_fields()]

        self.fill_in_optional = attrs.pop('_fill_optional', False)
        # error for non existing fields
        if isinstance(self.fill_in_optional, (tuple, list, set)):
            for field in self.fill_in_optional:
                if not field_exists(field):
                    raise AttributeError(
                        '_fill_optional field "%s" is not related to model "%s"'
                        % (field, self.model.__name__)
                    )
```

Please, tell me if makes any difference for you and feel free to use/adapt if you find that it makes sense.